test/SpokePool.Relay.ts

```diff
@@ -325,27 +345,345 @@ describe("SpokePool Relayer Logic", async function () {
+        await expect(
+          spokePool.connect(relayer).fillRelayUSSInternal(
+            relayExecution,
+            relayer.address,
+            false // isSlowFill
+          )
+        ).to.be.revertedWith("RelayFilled");
+      });
+      it("fast fill replacing speed up request emits correct FillType", async function () {
+        const relayExecution = await getRelayExecutionParams(relayData);
+        await spokePool.setFillStatus(relayExecution.relayHash, FillStatus.RequestedSlowFill);
+        await expect(
+          spokePool.connect(relayer).fillRelayUSSInternal(
+            relayExecution,
+            relayer.address,
+            false // isSlowFill
+          )
+        )
+          .to.emit(spokePool, "FilledUSSRelay")
+          .withArgs(
+            relayData.inputToken,
+            relayData.outputToken,
+            relayData.inputAmount,
+            relayData.outputAmount,
+            relayExecution.repaymentChainId,
+            relayData.originChainId,
+            relayData.depositId,
+            relayData.fillDeadline,
+            relayData.exclusivityDeadline,
+            relayData.exclusiveRelayer,
+            relayer.address,
+            relayData.depositor,
+            relayData.recipient,
+            relayData.message,
+            [
+              relayExecution.updatedRecipient,
+              relayExecution.updatedMessage,
+              relayExecution.updatedOutputAmount,
+              // Testing that this FillType is not "FastFill"
+              FillType.ReplacedSlowFill,
+            ]
+          );
+      });
+      it("slow fill emits correct FillType", async function () {
+        const relayExecution = await getRelayExecutionParams(relayData);
+        await destErc20.connect(relayer).transfer(spokePool.address, relayExecution.updatedOutputAmount);
+        await expect(
+          spokePool.connect(relayer).fillRelayUSSInternal(
+            relayExecution,
+            relayer.address,
+            true // isSlowFill
+          )
+        )
+          .to.emit(spokePool, "FilledUSSRelay")
+          .withArgs(
+            relayData.inputToken,
+            relayData.outputToken,
+            relayData.inputAmount,
+            relayData.outputAmount,
+            relayExecution.repaymentChainId,
+            relayData.originChainId,
+            relayData.depositId,
+            relayData.fillDeadline,
+            relayData.exclusivityDeadline,
+            relayData.exclusiveRelayer,
+            relayer.address,
+            relayData.depositor,
+            relayData.recipient,
+            relayData.message,
+            [
+              relayExecution.updatedRecipient,
+              relayExecution.updatedMessage,
+              relayExecution.updatedOutputAmount,
+              // Testing that this FillType is "SlowFill"
+              FillType.SlowFill,
+            ]
+          );
+      });
+      it("fast fill emits correct FillType", async function () {
```

Comment:
Just so I understand, does this fast fill _replace_ the slow fill? Or is this a fresh fill?

If it's fresh, it would be good to add a test to be sure that we can overwrite the slow fill enum with a filled enum.

Reply:
A fast fill can replace a slow fill request or be a fresh fill. there is a specific test for this on line 400: "fast fill replacing speed up request emits correct FillType", PTAL!